qiskit_dynamics/models/rotating_frame.py

```diff
@@ -354,20 +350,14 @@ def _conjugate_and_add(
         # get frame transformation matrix in diagonal basis
         # assumption that F is anti-Hermitian implies conjugation of
         # diagonal gives inversion
-        exp_freq = np.exp(self.frame_diag * t)
+        exp_freq = unp.exp(self.frame_diag * t)
         frame_mat = exp_freq.conj().reshape(self.dim, 1) * exp_freq
-        if issparse(out):
-            out = out.multiply(frame_mat)
-        elif type(out).__name__ == "BCOO":
-            out = out * frame_mat.data
-        else:
-            out = frame_mat * out
+
+        out = unp.multiply(out, frame_mat)
 
         if op_to_add_in_fb is not None:
-            if issparse(out):
-                op_to_add_in_fb = csr_matrix(op_to_add_in_fb)
-            elif type(out).__name__ == "BCOO":
-                op_to_add_in_fb = to_BCOO(op_to_add_in_fb)
+            if issparse(out) or type(out).__name__ == "BCOO":
+                op_to_add_in_fb = DYNAMICS_NUMPY_ALIAS(like=out).asarray(op_to_add_in_fb)
```

Comment:
Same here, I think this can just be:
```suggestion
            op_to_add_in_fb = DYNAMICS_NUMPY_ALIAS(like=out).asarray(op_to_add_in_fb)
```